File: src/main/java/no/nav/bidrag/beregn/forholdsmessigfordeling/beregning/ForholdsmessigFordelingBeregning.java

package no.nav.bidrag.beregn.forholdsmessigfordeling.beregning;

import java.util.List;
import no.nav.bidrag.beregn.forholdsmessigfordeling.bo.GrunnlagBeregningPeriodisert;
import no.nav.bidrag.beregn.forholdsmessigfordeling.bo.ResultatBeregning;

public interface ForholdsmessigFordelingBeregning {

  List<ResultatBeregning> beregn(GrunnlagBeregningPeriodisert grunnlagBeregningPeriodisert);

  static ForholdsmessigFordelingBeregning getInstance(){
    return new ForholdsmessigFordelingBeregningImpl();
  }

}

File: src/main/kotlin/no/nav/bidrag/beregn/forholdsmessigfordeling/beregning/ForholdsmessigFordelingBeregning.kt
package no.nav.bidrag.beregn.forholdsmessigfordeling.beregning

import no.nav.bidrag.beregn.forholdsmessigfordeling.bo.GrunnlagBeregningPeriodisert
import no.nav.bidrag.beregn.forholdsmessigfordeling.bo.ResultatBeregning

fun interface ForholdsmessigFordelingBeregning {
    fun beregn(grunnlag: GrunnlagBeregningPeriodisert): List<ResultatBeregning>

    companion object {
        fun getInstance(): ForholdsmessigFordelingBeregning {
            return ForholdsmessigFordelingBeregningImpl()
        }
    }
}
